add SimpleWebXR integration first draft

Rufus31415 opened the pull request:
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

[SimpleWebXR integration](https://github.com/Rufus31415/Simple-WebXR-Unity)

Rufus31415 committed:
add SimpleWebXR integration first draft

Commit:
Added Simple-WebXR-Unity submodule

Commit:
removed plugins from source control

added webxr plugin utility

Rufus31415 committed:
add WebXR profiles

Rufus31415 committed:
Add WebXR controllers support

Rufus31415 committed:
tuning WebXRController

Rufus31415 committed:
add hand controller first implementation

StephenHodgson committed:
Merge branch 'development' into feature/SimpleWebXR_integration

Rufus31415 committed:
Merge branch 'feature/SimpleWebXR_integration' of https://github.com/XRTK/WebXR into feature/SimpleWebXR_integration

SimonDarksideJ committed:
Upgraded the script and meta file issues in the current package. (#9)

* Upgraded the script and meta file issues in the current package.

* Resolved the Editor namespace fix

* Uplifted XRTK.Core dependencies

* Added WebXR Platform Service install profile

StephenHodgson commented on XRTK.WebXR/Packages/com.xrtk.webxr/package.json:
```diff
@@ -10,18 +10,17 @@
     "reality"
   ],
-  "version": "0.2.1",
+  "version": "0.2.2",
```
package version should be `0.2.0`

SimonDarksideJ replied:
I've actually reset this to 0.1.0, as it is the first iteration for the WebXR package.  Dependencies shouldn't need to follow the minor version of its parent.  Thoughts @StephenHodgson ?

StephenHodgson commented on XRTK.WebXR/Packages/manifest.json:
```diff
@@ -16,7 +16,7 @@
     "com.unity.textmeshpro": "2.1.3",
     "com.unity.ugui": "1.0.0",
-    "com.xrtk.core": "0.2.0",
+    "com.xrtk.core": "0.2.9",
```
remove this entry here, since the dependency is defined in the `package.json`

SimonDarksideJ replied:
Updated in latest patches

StephenHodgson commented on XRTK.WebXR/Packages/com.xrtk.webxr/package.json:
```diff
@@ -10,18 +10,17 @@
   "author": "XRTK Team (https://github.com/XRTK)",
   "dependencies": {
-    "com.xrtk.core": "0.2.1"
+    "com.xrtk.core": "0.2.9"
```
`xrtk.core` version could be updated to `0.2.11`
We should likely add your package dependency here as well.

SimonDarksideJ replied:
resolved

StephenHodgson reviewed (changes requested):
See comments

SimonDarksideJ committed:
Updated Submodule check-out
Renamed folder Profiles~/DataProviders -> Controllers as per XRTK standard

SimonDarksideJ committed:
Fixed Plug-in loader to use the new source file locations
Cleaned up implementation

SimonDarksideJ committed:
Updated submodule to Beta 1 release

StephenHodgson committed:
Change requests/simple webxr integration (#10)

* first round of changes

* updated license headers for webxr native bindings

* Added XRTK.WebXR.asmdef

updated access for some internal classes

* Added some notes about camera

* Make WebXRInputSource.TargetRayMode a property

* Removed WebXRPluginUtility

* a bit more formatting and cleanup

set fallback headheight to WebXRCameraDataProviderProfile.DefaultHeadHeight

* updated bindings with mutiview2 checks

* Added assembly infos

updated script icons

* updated license year

* updated readme with build status and open upm link

* updated webxr platform definition

* added packages-lock.json to gitifnore

* updated simple webxr submodule

* updated webxr

* added webxr build info

updated bindings with IsXrAvailable
Added preliminary hooks to enable/disable the unity xr pipeline (legacy)

StephenHodgson committed:
updated dependencies with latest release

SimonDarksideJ commented:
What's the current status now @StephenHodgson / @Rufus31415 , are we any closer?

StephenHodgson commented:
I haven't had time to dig into the single pass stuff, but it seems possible. Just need more free time to work on this ⌛

StephenHodgson committed:
updated project to 2019.4.30f1

updated xrtk.core pacakge

StephenHodgson committed:
added assembly infos

StephenHodgson committed:
GitHub Actions Workflow (#11)

* Create build.yml

* modernized the project

* fixed package installer

StephenHodgson committed:
Merge branch 'development' into feature/SimpleWebXR_integration

# Conflicts:
#	XRTK.WebXR/Packages/com.xrtk.webxr/Editor/AssemblyInfo.cs
#	XRTK.WebXR/Packages/com.xrtk.webxr/Editor/AssemblyInfo.cs.meta
#	XRTK.WebXR/Packages/com.xrtk.webxr/Editor/WebXRPackageInstaller.cs
#	XRTK.WebXR/Packages/com.xrtk.webxr/Runtime/AssemblyInfo.cs
#	XRTK.WebXR/Packages/com.xrtk.webxr/Runtime/AssemblyInfo.cs.meta
#	XRTK.WebXR/Packages/com.xrtk.webxr/package.json
